Subject: sched: Move rtsched_debug parameter into CONFIG_SMP
From: John Jolly <jjolly@suse.de>
Date: Tue Jun 10 18:29:00 CEST 2011
Patch-mainline: never, SUSE specific
References: bnc#882096

When building the SLES kernel with CONFIG_SMP unset, the build error
"‘sysctl_sched_rtsched_debug’ undeclared" occurs.

This is due to the declaration of 'sysctl_sched_rtsched_debug' within
CONFIG_SMP. This fix moves the parse_rtsched_debug function and
coresponding early_param into the same CONFIG_SMP as the declaration.

Signed-off-by: John Jolly <jjolly@suse.de>
---
 Documentation/kernel-parameters.txt |    3 ++-
 kernel/sched/rt.c                   |   14 +++++++-------
 2 files changed, 9 insertions(+), 8 deletions(-)

--- a/kernel/sched/rt.c
+++ b/kernel/sched/rt.c
@@ -702,6 +702,13 @@ static void __enable_runtime(struct rq *
 
 int __read_mostly sysctl_sched_rtsched_debug;
 
+static int __init parse_rtsched_debug(char *arg)
+{
+	sysctl_sched_rtsched_debug = 1;
+	return 0;
+}
+early_param("rtsched_debug", parse_rtsched_debug);
+
 static int balance_runtime(struct rt_rq *rt_rq)
 {
 	int more = 0;
@@ -2058,10 +2065,3 @@ void print_rt_stats(struct seq_file *m,
 	rcu_read_unlock();
 }
 #endif /* CONFIG_SCHED_DEBUG */
-
-static int __init parse_rtsched_debug(char *arg)
-{
-	sysctl_sched_rtsched_debug = 1;
-	return 0;
-}
-early_param("rtsched_debug", parse_rtsched_debug);
--- a/Documentation/kernel-parameters.txt
+++ b/Documentation/kernel-parameters.txt
@@ -2844,7 +2844,8 @@ bytes respectively. Such letter suffixes
 
 	sched_debug	[KNL] Enables verbose scheduler debug messages.
 
-	rtsched_debug   [KNL] Disallows borrowing of excess rt_runtime from neighboring runqueues.
+	rtsched_debug   [KNL] On systems with CONFIG_SMP set, disallows borrowing of excess
+			rt_runtime from neighboring runqueues.
 			  /proc/sys/kernel/sched_rt_runtime_us becomes a hard limit per runqueue.
 
 	nortsched       [KNL] Disables realtime task bandwidth control.
